File: src/main/java/com/classy/instagram/user/repository/UserRepository.java
package com.classy.instagram.user.repository;

import com.classy.instagram.user.entity.User;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.Optional;

@Repository
public interface UserRepository extends JpaRepository<User, String> {
    Optional<User> findByEmailAndPassword(String email, String password);

    List<User> findByNicknameContains(String nickname);

}

File: src/main/java/com/classy/instagram/user/repository/UserRepository.kt
package com.classy.instagram.user.repository

import com.classy.instagram.user.entity.User
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository


@Repository
interface UserRepository : JpaRepository<User, String> {
    fun findByNicknameContains(nickname: String): List<User>
}
